fire/SFMainMod.F90

```diff
@@ -902,37 +980,103 @@ subroutine  crown_damage ( currentSite )
+                height_cbb                           = currentCohort%hite - crown_depth
+                passive_crown_FI                     = 0.0_r8  
+                ignite_crown                         = 0.0_r8  
+                currentCohort%passive_crown_fire_flg = 0
 
-                endif
+                
+                ! Evaluate for passive crown fire ignition
+                if (EDPftvarcon_inst%crown_fire(currentCohort%pft) > 0.001_r8) then
+                   
+                   ! Note: crown_ignition_energy to be calculated based on foliar moisture content from FATES-Hydro
+                   ! EQ 3 Van Wagner 1977
+                   ! h = crown_ignite_energy (kJ/kg), m = foliar moisture content based on dry fuel (%)
+                   ! crown_ignite_energy = 460 + 26 * m
+
+                   ! Crown fuel ignition potential, EQ 8 Bessie and Johnson 1995
+                   ! Van Wagner EQ4 FI = (Czh)**3/2 where z=crown base height,h=heat crown ignite energy, FI=fire intensity
+                   ! 0.01 = C from Van Wagner 1977 EQ4 for crown of base height 6m, 100% FMC, and FI 2500kW/m
+                   passive_crown_FI = (0.01_r8 * height_cbb *EDPftvarcon_inst%crown_ignite_energy(currentCohort%pft))**1.5_r8
+
+                   if (currentPatch%FI >= crown_fire_threshold) then ! 200 kW/m = threshold for crown fire potential
+                      
+                      ! Initiation of passive crown fire, EQ 14 Bessie and Johnson 1995
+                      ignite_crown = currentPatch%FI/passive_crown_FI
+                      
+                      if (ignite_crown >= 1.0_r8) then
+                         currentCohort%passive_crown_fire_flg = 1 ! passive crown fire ignited
+                         write(fates_log(),*) 'SF currentCohort%passive_crown_fire_flg, height_cbb = ', currentCohort%passive_crown_fire_flg, height_cbb  ! slevis diag
+                         ! "...in passive crown fires and high intensity surface
+                         ! fires trees can survive. Jack, red
+                         ! and white pine survive...scars used in dating fires"
+                         ! Johnson, E.A. 1992 Fire and Veg Dynamics: North American boreal forest. Cambridge Press
+                         currentCohort%fraction_crown_burned = EDPftvarcon_inst%crown_fire(currentCohort%pft)
+
+                         ! Critical intensity for active crowning (kW/m)
+                         ! EQ 12 Bessie and Johnson 1995
+                         ! Fuels / 0.45 to get biomass but note that the 0.45
+                         ! cancels out and could be removed. Also dividing
+                         ! critical_mass_flow_rate by 3.34, an empirical
+                         ! constant in Bessie & Johnson 1995
+                         leaf_c = currentCohort%prt%GetState(leaf_organ, all_carbon_elements)
+                         currentPatch%fuel_bulkd = currentCohort%n * leaf_c /  &
```

Comment:
To make sure I understand...
In the code I see twigs as a component of the litter variable:
`litt_c%ag_cwd(1)`
Is this the variable that you have in mind?

Otherwise in subroutine InitPRTGlobalAllometricCarbon I see a list of organs, but no mention of twigs.